lib/block.ml

```diff
@@ -160,19 +161,64 @@ let rec error_padding = function
       let xs = error_padding xs in
       x :: xs
 
-let pp_errors ppf t =
+let compute_delimiter ?delim outputs =
+  let s =
+    Format.asprintf "%a" (Format.pp_print_list (Output.pp ~pad:0)) outputs
+  in
+  let is_inadequate delim =
+    Astring.String.is_infix ~affix:("]" ^ delim ^ "}") s
+  in
+  let rec loop ~base_delim n =
+    let delim =
+      match n with 0 -> base_delim | n -> Format.sprintf "%s_%d" base_delim n
+    in
+    if is_inadequate delim then loop ~base_delim (n + 1) else delim
+  in
+  match delim with
+  | None when is_inadequate "" -> Some (loop ~base_delim:"mdx_error" 0)
+  | None -> None
+  | Some base_delim -> Some (loop ~base_delim 0)
+
+let pp_error ?syntax ?delim ppf outputs =
+  match syntax with
+  | Some Syntax.Markdown ->
+      Fmt.pf ppf "```\n```mdx-error\n%a\n"
+        Fmt.(list ~sep:(any "\n") Output.pp)
+        outputs
+  | Some Syntax.Mli | Some Syntax.Mld ->
+      let delim = compute_delimiter ?delim outputs in
```

Comment:
You are modifying the wrong delimiter: the one given by the user, instead of `err` (or I would prefer the more explicit `mdx_error` that you suggested).

If `outputs` contains the delimiter given by the user, it breaks the syntax...

Reply:
Makes sense, should be good now.

Reply:
Yes!